### ISSUE TITLE
Add true multitouch trackpad support.

### ISSUE THREAD

**Issue by wusticality** [bevyengine/bevy#6174]
## What problem does this solve or what need does it fill?

At present, trackpad events are captured as if the trackpad were a mouse. For example, on a MacBook Pro, events come in as a left mouse press / release event. The MacBook Pro trackpad is actually a multitouch display, however. Bevy should emit touches as if it were any other multitouch display, along with press / release events.

## What solution would you like?

After talking to @alice-i-cecile, it seems the solution is to have Bevy emit touches (as if it were a multitouch display) _and_ emit the mouse behavior if someone wants to opt into that default behavior. An example of what I'd like to accomplish using the trackpad is:

* Pan the camera by dragging two fingers on the trackpad.
* Zoom in and out by detecting pinch to zoom gestures.
* Other multitouch gestures as things come up.

One potential issue is that the MacBook Pro trackpad touch events should support press / release states when your finger touches and releases from the screen, _and also trackpad click events as the trackpad itself can be fully depressed_. Both types of touches should be supported which might be an issue with the current `Touches` implementation:

```rust
pub struct Touches {
    /// A collection of every [`Touch`] that is currently being pressed.
    pressed: HashMap<u64, Touch>,
    /// A collection of every [`Touch`] that just got pressed.
    just_pressed: HashMap<u64, Touch>,
    /// A collection of every [`Touch`] that just got released.
    just_released: HashMap<u64, Touch>,
    /// A collection of every [`Touch`] that just got cancelled.
    just_cancelled: HashMap<u64, Touch>,
}
```

It may be that Bevy should have a `Trackpad` input abstraction for this case that supports both touch events _and_ button-click events in one interface but I'm not sure.

## What alternative(s) have you considered?

For now, I'm getting around this by using keyboard shortcuts to zoom the world and holding a modifier key while dragging my finger on the trackpad for drag events which is not ideal.

**Comment by mockersf** [comment]
Complete support for that needs to be done in winit: https://github.com/rust-windowing/winit/issues/2379. It's not really possible in Bevy without winit support.

Next version of winit will expose `TouchpadMagnify` and `TouchpadRotate` events (https://github.com/rust-windowing/winit/pull/2157). 2 fingers dragging already works and is detected as mouse wheel.

**Comment by wusticality** [comment]
Yeah, I wasn't sure if Bevy should directly support gestures as they're totally up to the user, but I guess there are some common ones (pinch to zoom, drag with two fingers, etc.).

**Comment by wusticality** [comment]
> Complete support for that needs to be done in winit: [rust-windowing/winit#2379](https://github.com/rust-windowing/winit/issues/2379). It's not really possible in Bevy without winit support.
> 
> Next version of winit will expose `TouchpadMagnify` and `TouchpadRotate` events ([rust-windowing/winit#2157](https://github.com/rust-windowing/winit/pull/2157)). 2 fingers dragging already works and is detected as mouse wheel.

Huh, I don't get any `MouseWheel` events when dragging with two fingers, at least on my MacBook Pro (or are you saying in `winit`, not Bevy)?

**Comment by mockersf** [comment]
I get them with Bevy on my Mac, with example `mouse_input_events`

**Comment by xgbwei** [comment]
> Huh, I don't get any `MouseWheel` events when dragging with two fingers, at least on my MacBook Pro (or are you saying in `winit`, not Bevy)?

Also works fine on my MBP. I don't see anything in system preferences that might break it, so this seems like it could be caused by another app?

**Comment by james7132** [comment]
Marking this as blocked until winit supports this.

**Comment by megaboy101** [comment]
> Complete support for that needs to be done in winit: [rust-windowing/winit#2379](https://github.com/rust-windowing/winit/issues/2379). It's not really possible in Bevy without winit support.
> 
> Next version of winit will expose `TouchpadMagnify` and `TouchpadRotate` events ([rust-windowing/winit#2157](https://github.com/rust-windowing/winit/pull/2157)). 2 fingers dragging already works and is detected as mouse wheel.

Hey, just stumbled upon this issue and it looks like winit now supports `TouchpadMagnify` and `TouchpadRotate` as of v0.28.0?

I'd be down to make a PR to support this in Bevy, although I'm not very familiar with the engine's internal's atm (I only started playing around with Bevy a few weeks ago 😅)

**Comment by alice-i-cecile** [comment]
I believe those two events were added in https://github.com/bevyengine/bevy/pull/8791 :)

**Comment by simbleau** [comment]
@james7132  @alice-i-cecile  This is no longer blocked

**Comment by JeanMertz** [comment]
> @james7132 @alice-i-cecile This is no longer blocked

Is it not? I was looking for a way to support two-finger drag, but Winit's [features page](https://github.com/rust-windowing/winit/blob/4a8648be571076945a030fd18f4507a275d49967/FEATURES.md#input-handling-1) still mentions multitouch as not implemented for macOS.

**Comment by simbleau** [comment]
It seems like they maybe just forgot to update the feature matrix:
https://github.com/rust-windowing/winit/pull/2157
